src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -1016,6 +1017,98 @@ class CreateIndexStatementSegment(BaseSegment):
     )
 
 
+class CreateColumnstoreIndexStatementSegment(BaseSegment):
+    """A `CREATE COLUMNSTORE INDEX` statement.
+
+    https://learn.microsoft.com/en-us/sql/t-sql/statements/create-columnstore-index-transact-sql?view=sql-server-ver15
+    """
+
+    type = "create_columnstore_index_statement"
+    _on_partitions = Sequence(
```

Comment:
The repo's code style is just putting these inline so `match_grammar` just reads top to bottom. You could put comments above certain sections if you wish.

Reply:
Moved the _on_partitions and _with_option into match_grammar.